stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/EventHandlerImpl.kt

```diff
@@ -65,65 +65,109 @@ import io.getstream.chat.android.offline.experimental.extensions.state
 import io.getstream.chat.android.offline.extensions.mergeReactions
 import io.getstream.chat.android.offline.extensions.setMember
 import io.getstream.chat.android.offline.extensions.updateReads
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 internal class EventHandlerImpl(
     private val domainImpl: ChatDomainImpl,
     private val client: ChatClient,
 ) {
+    private val eventBuffer = MutableSharedFlow<List<ChatEvent>>()
+    private val connectEventBuffer = MutableSharedFlow<ChatEvent>(
```

Comment:
This is optional, but the idea behind this is that these events should replace one another if they come in quick succession (e.g. lost connection but quickly reconnected). This way in the **probably rare** case of a busy consumer (collector) there will not be a case where we are still processing old and irrelevant values.